### PR TITLE
[XLA:GPU] Add cuDNN sliding window attention support

### PR DESCRIPTION
**Pull request by Cjkkkk**
* Add support for cuDNN sliding window attention in XLA runner, not supported in pattern matcher.
* required by Gemma 2 workload.

### CODE REVIEW

**Review by akuegel — COMMENTED** [review]
As far as I understand, this PR only adds sliding window support for custom calls, not for the pattern matcher. Can you please adjust the description accordingly?

File: xla/service/gpu/backend_configs.proto [hunk 1]
```diff
@@ -261,6 +261,9 @@ message CudnnfMHABackendConfig {
 
   // Whether force deterministic for bwd
   bool force_deterministic = 23;
+
+  // Sliding window length
```
[the remark this answers is not in the document]
Comment (Cjkkkk) [reply xla/service/gpu/backend_configs.proto:265]: added.

**Comment by Cjkkkk** [comment]
> As far as I understand, this PR only adds sliding window support for custom calls, not for the pattern matcher. Can you please adjust the description accordingly?

that's correct, i updated the description accordingy.

**Review by akuegel — APPROVED** [review]

File: xla/service/gpu/tests/gpu_fused_mha_test.cc [hunk 2]
```diff
@@ -1102,6 +1102,177 @@ class FlashAttentionBMMScalePaddingMaskSoftmaxBMM
...
+    )";
+    return hlo_text;
+  }
+
+  template <typename T>
+  void TestImpl_Flash_Attention_Training_BMM1_SlidingWindowMask_Softmax_BMM2() {
+    if (skip_reason_) GTEST_SKIP() << *skip_reason_;
+    if (GetDnnVersionInfoOrDefault(backend().default_stream_executor()) <
+        se::dnn::VersionInfo(9, 0, 0)) {
+      GTEST_SKIP() << "Flash Attention requires cuDNN >= 9.0.0.";
```
Comment (akuegel) [inline xla/service/gpu/tests/gpu_fused_mha_test.cc:1260]: It seems this needs to check for cudnn 9.2.0. We get this error message internally:

For cuDNN version below 9.2.0, sliding window attention is not supported

**Comment by loislo** [comment]
The tests are failing with the error

For cuDNN version below 9.2.0, sliding window attention is not supported

from

**Comment by Cjkkkk** [comment]
> The tests are failing with the error
> 
> For cuDNN version below 9.2.0, sliding window attention is not supported

Sorry about the confusion, it is fixed now.

**Review by akuegel — APPROVED** [review]

**Pull request closed without merging**